test/joiner_team_index_by_field_test.rb

```diff
@@ -33,7 +33,6 @@ def add_team_member(member_hash, private: false)
     def impl
       joiner_impl = JoinerImpl.new @site
       joiner_impl.data.merge! joiner_impl.collection_data
-      joiner_impl.promote_or_remove_data
```

Comment:
Because this was causing the tests to always pass; taking this line out, the tests failed until I performed the refactoring to get them to pass again.